riscv-pages/src/sequential_pages.rs

```diff
@@ -236,6 +243,44 @@ impl<S: State, I: Iterator<Item = Page<S>>> fmt::Debug for Error<S, I> {
     }
 }
 
+pub struct SeqPageChunkIter<S: State> {
+    pages: SequentialPages<S>,
+    n: u64,
+}
+
+impl<S: State> Iterator for SeqPageChunkIter<S> {
+    type Item = SequentialPages<S>;
+
+    fn next(&mut self) -> Option<SequentialPages<S>> {
+        if self.pages.count == 0 {
+            return None;
+        }
+        let addr = self.pages.addr;
+        let page_size = self.pages.page_size;
+        let count = if self.pages.count > self.n {
+            self.n
+        } else {
+            self.pages.count
+        };
+        self.pages.addr = self
+            .pages
+            .addr
+            .checked_add_pages_with_size(count, page_size)
+            .unwrap();
+        self.pages.count -= count;
+        // Safe because `self` owns the memory, and we're creating a sequence from pages that
+        // have been removed.
+        unsafe { Some(SequentialPages::from_mem_range(addr, page_size, count).unwrap()) }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let count = ((self.pages.count + self.n - 1) / self.n) as usize;
```

Comment:
Should we check for division by zero (self.n == 0) here?

Reply:
Good catch, as the n can be the result of some dynamic calculation and not the static use we do now. Will prefent chunks of zero (that would be an infinite loop) to be created.

Reply:
Done.